assets/js/components/surveys/CurrentSurvey.test.js

```diff
@@ -80,6 +80,120 @@ describe( 'CurrentSurvey', () => {
+				triggerID: 'jestSurvey',
+			} );
+
+		fetchMock.post(
+			/^\/google-site-kit\/v1\/core\/user\/data\/survey-event/,
+			{ body: {}, status: 200 }
+		);
+
+		const {
+			getByText,
+			getByPlaceholderText,
+			getByRole,
+			getByDisplayValue,
+			findByText,
+		} = render( <CurrentSurvey />, {
+			registry,
+		} );
+
+		expect( fetchMock ).toHaveFetched(
+			/^\/google-site-kit\/v1\/core\/user\/data\/survey-event/
+		);
+
+		expect( fetchMock ).toHaveBeenCalledTimes( 1 );
+
+		// Submit button should be disabled if text input is empty
+		expect( getByRole( 'button', { name: 'Submit' } ) ).toHaveAttribute(
+			'disabled'
+		);
+
+		// Check question_text is set
+		expect(
+			getByText( 'How satisfied are you with Site Kit?' )
+		).toBeInTheDocument();
+
+		// Check subtitle is set
+		expect(
+			getByText( 'Based on your experience so far, tell us.' )
+		).toBeInTheDocument();
+
+		const STRING_100_CHARACTERS =
+			'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec suscipit auctor dui, id faucibus nisl';
+
+		const STRING_110_CHARACTERS = STRING_100_CHARACTERS + ' rhoncus n';
+
+		// Now enter answer text
+		fireEvent.change( getByPlaceholderText( 'Write here' ), {
+			target: { value: STRING_110_CHARACTERS },
+		} );
+
+		// Input does not have a label so this is the best way to assert the text has been entered correctly
+		expect(
+			getByDisplayValue( STRING_100_CHARACTERS )
+		).toBeInTheDocument();
+
+		// Submit button should be enabled if text has been entered
+		expect( getByRole( 'button', { name: 'Submit' } ) ).not.toHaveAttribute(
+			'disabled'
+		);
+
+		// check clearing and entering input again
+		fireEvent.change( getByPlaceholderText( 'Write here' ), {
+			target: { value: '' },
+		} );
+		expect( getByRole( 'button', { name: 'Submit' } ) ).toHaveAttribute(
+			'disabled'
+		);
+		fireEvent.change( getByPlaceholderText( 'Write here' ), {
+			target: { value: STRING_110_CHARACTERS },
+		} );
+		expect( getByRole( 'button', { name: 'Submit' } ) ).not.toHaveAttribute(
+			'disabled'
+		);
+
+		// Now submit question
+		fireEvent.click( getByRole( 'button', { name: 'Submit' } ) );
+
+		expect( fetchMock ).toHaveBeenCalledTimes( 2 );
+
+		expect( fetchMock ).toHaveFetched(
+			'/google-site-kit/v1/core/user/data/survey-event?_locale=user',
+			{
+				body: {
+					data: {
+						event: {
+							question_answered: {
+								question_ordinal: 1,
+								answer: {
+									answer:
+										'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec suscipit auctor dui, id faucibus nisl',
+								},
+							},
+						},
+						session: {
+							session_id: 'storybook_session',
+							session_token: 'token_12345',
+						},
+					},
+				},
+				credentials: 'include',
+				headers: {
+					Accept: 'application/json, */*;q=0.1',
+					'Content-Type': 'application/json',
+				},
+				method: 'POST',
+			}
+		);
+
+		await findByText( 'Thanks for sharing your thoughts!' );
```

Comment:
I know this should fail if the `await` doesn't fulfill, but that's a bit less obvious than an expectation around this… maybe:
```suggestion
		const completionText = await findByText( 'Thanks for sharing your thoughts!' );
		expect( completionText ).not.toBeEmpty();
```
or something like that?

Reply:
It's actually to stop Act warnings.  If we don't wait for this then I think we get Act warnings.  This is actually copied from another test though!  `should send a 'question_answered' event when a question is answered`, so for consistency I think we should leave it (I also have something similar on my next two open PRs) :smile: